File: problems/56/solution.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if not intervals:
            return []
        
        intervals.sort(key=lambda x: x[0])
        
        for i in range(1, len(intervals)):
            if intervals[i][0] <= intervals[i - 1][1]:
                intervals[i] = [min(intervals[i][0], intervals[i - 1][0]), max(intervals[i][1], intervals[i - 1][1])]
                intervals[i - 1] = None
        
        return [i for i in intervals if i]

File: problems/56/solution.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        if (empty(intervals)) {
            return {};
        }
        
        vector<vector<int>> ans;
        
        sort(begin(intervals), end(intervals));
        
        for (const vector<int>& interval: intervals) {
            if (empty(ans) or interval.front() > ans.back().back()) {
                ans.push_back(interval);
            } else {
                ans.back().back() = max(ans.back().back(), interval.back());
            }
        }
        
        return ans;
    }
};